packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -155,28 +149,46 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
   },
 
   getStatus(version: DocumentVersion, otherDocumentStatuses?: DocumentMetadata['availableStatus']) {
+    let draftVersion: DocumentVersion | undefined;
+    let publishedVersion: DocumentVersion | undefined;
+
     const isDraft = version.publishedAt === null;
 
+    if (isDraft) {
+      draftVersion = version;
+    } else {
+      publishedVersion = version;
+    }
+
+    if (otherDocumentStatuses && otherDocumentStatuses.at(0)) {
+      const otherVersion = otherDocumentStatuses.at(0);
+      if (otherVersion?.publishedAt) {
+        publishedVersion = otherVersion;
+      } else {
+        draftVersion = otherVersion;
+      }
+    }
+
     if (!otherDocumentStatuses?.length) {
-      // It there are no other versions we take the current version status
+      // If there are no other versions, the current version is the latest
       return isDraft ? CONTENT_MANAGER_STATUS.DRAFT : CONTENT_MANAGER_STATUS.PUBLISHED;
     }
```

Comment:
nice, was toying with it, could we simplify a bit the comparison to check if there is a draft version or published version available?
```ts
    if (isDraft) {
      draftVersion = version;
    } else {
      publishedVersion = version;
    }
	
     const otherVersion = otherDocumentStatuses?.at(0);
      if (otherVersion?.publishedAt) {
        publishedVersion = otherVersion;
      } else if (otherVersion) {
        draftVersion = otherVersion;
      }

    if(!draftVersion) return published
    if (!publishedVersion) return draft
    
    const isDraftModified = getIsVersionLatestModification(draftVersion, publishedVersion);
    return isDraftModified ? CONTENT_MANAGER_STATUS.MODIFIED : CONTENT_MANAGER_STATUS.PUBLISHED;
```